### PR TITLE
Move MidiHandler class to internal header in midi module

### PR DESCRIPTION
**Pull request by dreamer**
Mostly a cleanup, fixing several warnings and tidying-up interface.

This is final cleanup PR in preparation for #262.

### CODE REVIEW

File: src/midi/midi_coremidi.h [hunk 1]
```diff
@@ -22,20 +22,27 @@
...
 private:
 	MIDIPortRef m_port;
 	MIDIClientRef m_client;
 	MIDIEndpointRef m_endpoint;
 	MIDIPacket* m_pCurPacket;
 public:
-	MidiHandler_coremidi()  {m_pCurPacket = 0;}
-	const char * GetName(void) { return "coremidi"; }
-	bool Open(const char * conf) {	
+	MidiHandler_coremidi() { m_pCurPacket = 0; }
```
Comment (kcgen) [inline src/midi/midi_coremidi.h:40]: Can this member assignment during construction `{ m_pCurPacket = 0; }` be done directly at the member definition?

File: src/midi/midi_handler.h [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+	virtual void Close() {}
+
+	virtual void PlayMsg(MAYBE_UNUSED const uint8_t *msg) {}
+
+	virtual void PlaySysex(MAYBE_UNUSED uint8_t *sysex, MAYBE_UNUSED size_t len) {}
+
+	virtual void ListAll(MAYBE_UNUSED Program *base) {}
+
+	MidiHandler *next;
```
Comment (kcgen) [inline src/midi/midi_handler.h:56]: maybe initialize to `nullptr`?

Comment (dreamer) [reply src/midi/midi_handler.h:56]: It is already being initialized in initializer list (not with `nullptr`), so let's keep it as it is.

File: src/midi/midi_coremidi.h (hunk 1 above)
Comment (dreamer) [reply src/midi/midi_coremidi.h]: I added one new commit to the PR to correctly initialize bot macOS-specific MIDI handlers :)

**Comment by dreamer** [comment]
Thanks :)

**Pull request merged**